aten/src/ATen/core/TensorImpl.h

```diff
@@ -1438,6 +1502,7 @@ struct CAFFE2_API TensorImpl : public c10::intrusive_ptr_target {
   bool is_contiguous_ = true;
   bool is_variable_ = false;
   bool is_wrapped_number_ = false;
+  bool allow_size_or_storage_change_ = true;
```

Comment:
While, on the face of it, the meaning of this field seems "obvious" (it controls whether or not to allow size or storage change), what the current documentation doesn't motivate *why* we need this field, and under what circumstances we disallow a size or storage change. That would be a very good comment to add here. Look to have the same amount of specificity as in: https://github.com/pytorch/pytorch/blob/a2e1d7f9b372fae1eaba5c6b3f165b28b3a10d82/aten/src/ATen/core/TensorImpl.h#L493

Reply:
In particular, when I see use sites of this boolean, they always seem to say something like, "You cannot do X for a variable that was detached." Is that the *real* semantic payload of this member? Then it should say so! (For example, I am not allowed to set coalesced on a tensor that does not "allow size or storage change". So the name of this field isn't even right, not at least how it is used, anyhow.)

Reply:
Some other things to think about when writing the docs: if I am a future developer and I'm looking to add a new field to tensor, when should I respect this field, versus not respect it? What's the criteria I should use?

Reply:
I renamed the flag to `is_created_from_data_or_detach_` to better reflect its actual semantics, and also add comments to explain why we need it and its usage.